.github/workflows/label-pr.yml

```diff
@@ -0,0 +1,64 @@
+name: Label PR if diff files contain a certain pattern
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+
+jobs:
+  label-if-contains-pattern:
+    runs-on: ubuntu-latest
+
+    permissions:
+      pull-requests: write
+      contents: read
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Get list of files changed in the PR
+        id: file_list
+        run: |
+          FILES=$(git diff --name-only origin/${{ github.event.pull_request.base.ref }}...HEAD)
+          echo "FILES<<EOF" >> $GITHUB_OUTPUT
+          echo "$FILES" >> $GITHUB_OUTPUT
+          echo "EOF" >> $GITHUB_OUTPUT
+
+      - name: Search if pattern is in modified files
+        id: grep_check
+        run: |
+          FILES="${{ steps.file_list.outputs.FILES }}"
+          FOUND=false
+
+          # These functions return a different backend between local and staging/production
+          # Changes using these functions should be tested on staging before merging
+          PATTERN="get_blob_db|get_email_configuration"
```

Comment:
Will any PR that modifies a file containing this pattern will get labeled, even if the modifications did not change a line that matches the pattern?

Maye consider using `git log -G'get_blob_db|get_email_configuration'` instead.

Reply:
> any PR that modifies a file containing this pattern will get labeled, even if the modifications did not change a line that matches the pattern?
yes. This does have the downside of having false positives.

But triggering only if the line matching the pattern is changed would lead to false negatives. It's likely that the pattern matched line doesn't change, but interaction with the db is changed.

Reply:
I'm worried that it will not be obvious why the label was added.

Particularly for `get_blob_db`, you [changed](https://github.com/dimagi/commcare-hq/pull/36377) the blob db so the issue you ran into would no longer happen, correct? Would you agree that that one is probably not necessary anymore.

I think it's a better idea to encourage devs to add more thorough test coverage to their PRs.

Reply:
Yes, i made changes to blob db so that specific problem wouldn't happen before. The intention behind this was to potentially catch any other unknown gaps, or prompt the dev to be aware of this potential.

> I'm worried that it will not be obvious why the label was added. I think it's a better idea to encourage devs to add more thorough automated test coverage to their PRs rather than prompting for manual testing on staging.

Both are fair points. Two alternative options instead is to flag the PR as medium risk (instead of "test on staging") or label it as "backend-env-risk" as more of a warning rather than a directive to do manual testing?

Reply:
I won't speak for `get_email_configuration`, but I think the blob db implementations are close enough to feature parity that an amorphous testing or risk directive is not likely to be helpful. If feature parity issues arise in the blob db, we should address them specifically, as you did in this case, rather than raising flags that are likely to have a lot of false positives and therefore be ignored.

Reply:
I'm not sure if this is possible - but this would remove false positives. 

What if it flags the PR only if
- the pattern is changed in the diff
- the variable assigned to the pattern is changed in the diff. For example, `blobdb = get_blob_db()` then it'll flag the PR if a diff within that file includes `blobdb`